pkg/watch/watch.go

```diff
@@ -358,15 +540,30 @@ func processEvents(changedFiles, deletedPaths []string, parameters WatchParamete
 		RandomPorts:              parameters.RandomPorts,
 		ErrOut:                   parameters.ErrOut,
 	}
-	err := parameters.DevfileWatchHandler(pushParams, parameters)
+	oldStatus := *componentStatus
+	err := parameters.DevfileWatchHandler(pushParams, parameters, componentStatus)
 	if err != nil {
-		// Log and output, but intentionally not exiting on error here.
-		// We don't want to break watch when push failed, it might be fixed with the next change.
+		if isFatal(err) {
+			return nil, err
+		}
 		klog.V(4).Infof("Error from Push: %v", err)
-		fmt.Fprintf(out, "%s - %s\n\n", PushErrorString, err.Error())
-	} else {
+		if parameters.WatchFiles {
+			// Log and output, but intentionally not exiting on error here.
+			// We don't want to break watch when push failed, it might be fixed with the next change.
+			fmt.Fprintf(out, "%s - %s\n\n", PushErrorString, err.Error())
```

Comment:
> If we're waiting for something to change here, it will help to print that in the output so that user knows what to do in case of an error.

The error is displayed, I think it is enough for the user to know something has to be fixed.

Reply:
> Why does the curl command still return an output even when I delete the service, or even while a deployment is being created after I delete it?

The port forward is done directly on the pod. I'm pretty sure the Service is not necessary anymore.

The time the previous pod is being terminated, you will get responses from it. You 'll get an error after the previous pod is terminated.